contrib/valkey-go/valkey.go

```diff
@@ -0,0 +1,347 @@
+	commands := commander.Commands()
+	if len(commands) == 0 {
+		return "", "", 0
+	}
+	command = commands[0]
+	statement = strings.Join(commands, "\n")
+	return command, statement, len(statement)
+}
+
+func processMultiCmds(multi []commander) (command, statement string, size int) {
+	var commands []string
+	var statements []string
+	for _, cmd := range multi {
+		cmdStr, stmt, cmdSize := processCmd(cmd)
+		size += cmdSize
+		commands = append(commands, cmdStr)
+		statements = append(statements, stmt)
+	}
+	command = strings.Join(commands, " ")
+	statement = strings.Join(statements, "\n")
+	return command, statement, size
+}
+
+func processMultiCompleted(multi ...valkey.Completed) (command, statement string, size int) {
+	cmds := make([]commander, len(multi))
+	for i, cmd := range multi {
+		cmds[i] = &cmd
+	}
+	return processMultiCmds(cmds)
+}
+
+func processMultiCacheableTTL(multi ...valkey.CacheableTTL) (command, statement string, size int) {
+	cmds := make([]commander, len(multi))
+	for i, cmd := range multi {
+		cmds[i] = &cmd.Cmd
+	}
+	return processMultiCmds(cmds)
+}
+
+func firstError(s []valkey.ValkeyResult) error {
+	for _, result := range s {
+		if err := result.Error(); err != nil && !valkey.IsValkeyNil(err) {
+			return err
+		}
+	}
+	return nil
+}
+
+func setClientCacheTags(s tracer.Span, result valkey.ValkeyResult) {
+	s.SetTag(ext.ValkeyClientCacheHit, result.IsCacheHit())
+	s.SetTag(ext.ValkeyClientCacheTTL, result.CacheTTL())
+	s.SetTag(ext.ValkeyClientCachePTTL, result.CachePTTL())
+	s.SetTag(ext.ValkeyClientCachePXAT, result.CachePXAT())
+}
+
+type buildStartSpanOptionsInput struct {
+	command        string
+	statement      string
+	size           int
+	isWrite        bool
+	isBlock        bool
+	isMulti        bool
+	isStream       bool
+	skipRawCommand bool
+}
+
+func (c *coreClient) buildStartSpanOptions(input buildStartSpanOptionsInput) []tracer.StartSpanOption {
+	opts := []tracer.StartSpanOption{
+		tracer.SpanType(ext.SpanTypeValkey),
+		tracer.ServiceName(c.clientConfig.serviceName),
+		tracer.Tag(ext.TargetHost, c.host),
+		tracer.Tag(ext.TargetPort, c.port),
+		tracer.Tag(ext.ValkeyClientVersion, valkey.LibVer),
+		tracer.Tag(ext.ValkeyClientName, valkey.LibName),
+		tracer.Tag(ext.Component, componentName),
+		tracer.Tag(ext.SpanKind, ext.SpanKindClient),
+		tracer.Tag(ext.DBType, ext.DBSystemValkey),
+		tracer.Tag(ext.DBSystem, ext.DBSystemValkey),
+		tracer.Tag(ext.ValkeyDatabaseIndex, c.option.SelectDB),
+		tracer.Tag(ext.ValkeyClientCommandWrite, input.isWrite),
+		tracer.Tag(ext.ValkeyClientCommandBlock, input.isBlock),
+		tracer.Tag(ext.ValkeyClientCommandMulti, input.isMulti),
+		tracer.Tag(ext.ValkeyClientCommandStream, input.isStream),
+		tracer.Tag(ext.ValkeyClientCommandWithPassword, c.option.Password != ""),
+	}
+	if input.command != "" {
+		opts = append(opts, []tracer.StartSpanOption{
+			// valkeyotel tags
+			tracer.Tag("db.stmt_size", input.size),
+			tracer.Tag("db.operation", input.command),
```

Comment:
Set same tags of valkeyotel
https://github.com/valkey-io/valkey-go/blob/92107ce9280700f370c2d9341ac12212692778d5/valkeyotel/trace.go#L375-L378